src/core/ledger/ledger.js

```diff
@@ -159,6 +165,47 @@ export class Ledger {
     return this.account(identityId);
   }
 
+  /**
+   * Get the Allocation associated with a particular Allocation ID.
+   *
+   * If the ID is not in the ledger, an error is thrown.
+   */
+  allocation(id: UuidT): Allocation {
+    const allocation = this._allocations.get(id);
+    if (allocation == null) throw new Error(`no Allocation for id: ${id}`);
+    return allocation;
+  }
+
+  /**
+   * Get the Distribution associated with a particular Distribution ID.
+   *
+   * If the ID is not in the ledger, an error is thrown.
+   */
+  distribution(id: UuidT): Distribution {
+    const distribution = this._distributions.get(id);
+    if (distribution == null) throw new Error(`no Distribution for id: ${id}`);
+    return distribution;
+  }
+
+  /**
+   * Get the Distribution associated with a particular Allocation ID.
+   *
+   * If the Allocation ID is not associated with a distribution, an error is thrown.
+   *
+   * This is less efficient than the `distribution` method.
+   */
+  distributionByAllocationId(allocationId: UuidT): Distribution {
+    for (const distribution of this._distributions.values()) {
+      if (
+        distribution.allocations.find(
+          (allocation) => allocation.id === allocationId
+        )
+      )
+        return distribution;
+    }
+    throw new Error(`no Distribution for allocation id: ${allocationId}`);
```

Comment:
I think we should invest upfront in making this performant, by adding a `_allocationIdToDistribution: Map<AllocationId, DistributionId>` field to the ledger.